lib/helpers/logger.ts

```diff
@@ -0,0 +1,65 @@
+import _debug from 'debug';
+import {
+  createLogger as _createLogger,
+  format,
+  transports,
+  type Logform,
+  type Logger,
+} from 'winston';
+import Transport from 'winston-transport';
+
+type LogInfo = Logform.TransformableInfo & {
+  label: string;
+  timestamp: string;
+};
+
+const debug = _debug('ember-this-fallback-plugin');
+
+class DebugTransport extends Transport {
+  public override log(info: LogInfo, next: () => void): void {
+    debug(info[Symbol.for('message')]);
+    next();
+  }
+}
+
+function concatMessage(label: string, message: string): string {
+  return `${label}\n\t${message}`;
+}
+
+const logFormatter = format.printf(({ level, label, timestamp, message }) => {
+  return `${String(timestamp)} [${level}] ${concatMessage(
+    String(label),
+    String(message)
+  )}`;
+});
+
+const debugFormatter = format.printf(({ label, message }) => {
+  return concatMessage(String(label), String(message));
+});
+
+export default function createLogger(name: string, label: string): Logger {
+  return _createLogger({
+    level: 'debug',
+    transports: [
+      new transports.File({
```

Comment:
Sample from `ember-this-fallback-plugin.log`:

```
2023-04-18T23:38:23.972Z [warn] /Users/krystanhuffmenne/Code/ember-this-fallback/dummy/tests/integration/plugin/mustache-statement-test.ts
	Found ambiguous mustache statement as attribute node value in `id={{property}}`. Falling back to runtime dynamic resolution. You can avoid this fallback by explicitly invoking a known helper with parens: `id={{(property)}}` or by prefacing a known property on `this` with `this`: `id={{this.property}}`. See https://github.com/tildeio/ember-this-fallback#embroider-compatibility for more details.
```

Reply:
I think it's good to emit a warning per usage, but maybe only emit the whole megillah once per warning type, at the end?

Reply:
Seems fine to hold them toward the end, but I liked the idea of having the actual answer copy-pasta-able, which necessitates logging once for each instance of the issue.